redisgraph/graph.py

```diff
@@ -121,16 +145,40 @@ def query(self, q, params=None, timeout=None):
         """
         Executes a query against the graph.
         """
+
+        # maintain original 'q'
+        query = q
+
+        # handle query parameters
         if params is not None:
-            q = self.build_params_header(params) + q
+            query = self.build_params_header(params) + query
+
+        # construct query command
+        # ask for compact result-set format
+        # specify known graph version
+        command = ["GRAPH.QUERY", self.name, query, "--compact", "version", self.version]
 
-        command = ["GRAPH.QUERY", self.name, q, "--compact"]
+        # include timeout is specified
         if timeout:
             if not isinstance(timeout, int):
                 raise Exception("Timeout argument must be a positive integer")
             command += ["timeout", timeout]
-        response = self.redis_con.execute_command(*command)
-        return QueryResult(self, response)
+
+        # issue query
+        try:
+            response = self.redis_con.execute_command(*command)
+            return QueryResult(self, response)
+        except redis.exceptions.ResponseError as e:
+            if "wrong number of arguments" in str(e):
+                print ("Note: RedisGraph Python requires server version 2.2.8 or above")
```

Comment:
for now, this is ok
if we introduce new flags there should be a module version check, so each flag is verified with its version and the query will not run if we have a version mismatch